datastore/sample.rb

```diff
@@ -0,0 +1,748 @@
+# Copyright 2016 Google, Inc
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+require "gcloud"
+
+def incomplete_key
+  gcloud = Gcloud.new
+  datastore = gcloud.datastore
+
+  # [START incomplete_key]
+  task_key = datastore.key "Task"
+  # [END incomplete_key]
+end
+
+def named_key
+  gcloud = Gcloud.new
+  datastore = gcloud.datastore
+
+  # [START named_key]
+  task_key = datastore.key "Task", "sampleTask"
+  # [END named_key]
+end
+
+def key_with_parent
+  gcloud = Gcloud.new
+  datastore = gcloud.datastore
+
+  # [START key_with_parent]
+  task_key = datastore.key "Task", "sampleTask"
+  task_key.parent = datastore.key "TaskList", "default"
+  # [END key_with_parent]
+
+  task_key
+end
+
+def key_with_multilevel_parent
+  gcloud = Gcloud.new
+  datastore = gcloud.datastore
+
+  # [START key_with_multilevel_parent]
+  user_key = datastore.key "User", "alice"
+  task_list_key = datastore.key "TaskList", "default"
+  task_key = datastore.key "Task", "sampleTask"
+  task_list_key.parent = user_key
+  task_key.parent = task_list_key
+  # [END key_with_multilevel_parent]
+
+  task_key
+end
+
+def entity_with_parent
+  gcloud = Gcloud.new
+  datastore = gcloud.datastore
+
+  # [START entity_with_parent]
+  task_key = datastore.key "Task", "sampleTask"
+  task_key.parent = datastore.key "TaskList", "default"
+
+  task = datastore.entity task_key do |t|
+    t["type"] = "Personal"
+    t["done"] = true
+    t["priority"] = 4
+    t["description"] = "Learn Cloud Datastore"
+  end
+  # [END entity_with_parent]
+
+  task
+end
+
+def properties
+  gcloud = Gcloud.new
+  datastore = gcloud.datastore
+
+  # [START properties]
+  task = datastore.entity "Task" do |t|
+    t["type"] = "Personal"
+    t["created"] = Time.now
+    t["done"] = true
+    t["priority"] = 4
+    t["percent_complete"] = 10.0
+    t["description"] = "Learn Cloud Datastore"
+  end
+  # [END properties]
+end
+
+def array_value
+  gcloud = Gcloud.new
+  datastore = gcloud.datastore
+
+  # [START array_value]
+  task = datastore.entity "Task", "sampleTask" do |t|
+    t["tags"] = %w(fun programming)
```

Comment:
Oooooooooh I wonder whether or not to use `%w(  )` for our arrays in samples.  It's a wonderful idiom, but I'm tempted to think that a fully written out (non-string) array would be more accessible for our samples.

I'd like to use and prefer `t["tags"] = ["fun", "programming"]` for our samples

*Misc: we do this for the samples for [`Entity#exclude_from_indexes`](https://googlecloudplatform.github.io/gcloud-ruby/#/docs/v0.7.2/gcloud/datastore/entity?method=exclude_from_indexes!-instance)*